src/Servers/Kestrel/Transport.Quic/src/WebHostBuilderQuicExtensions.cs

```diff
@@ -44,4 +44,18 @@ public static IWebHostBuilder UseQuic(this IWebHostBuilder hostBuilder, Action<Q
             services.Configure(configureOptions);
         });
     }
+
+    private static bool IsQuicSupported()
+    {
+        try
+        {
+            return QuicImplementationProviders.Default.IsSupported;
+        }
+        catch (PlatformNotSupportedException)
```

Comment:
IsSupported properties shouldn't throw. This seems like a bug that should be fixed in runtime. @ManickaP

Reply:
It's not that the individual property getter throws. It's that, on unsupported platforms, the **entire** System.Net.Quic.dll assembly is a stub composed purely of methods that just throw `PlatformNotSupportedException`.

Reply:
It would need to be quite a few properties and methods:

 * System.Net.Quic.QuicImplementationProviders::get_Default
 * System.Net.Quic.QuicImplementationProviders::get_MsQuic
 * System.Net.Quic.ImplementationsQuicImplementationProvider::ctor
 * System.Net.Quic.Implementations.MsQuic.MsQuicImplementationProvider::ctor
 * System.Net.Quic.Implementations.MsQuic.MsQuicImplementationProvider::get_IsSupported
 * System.Net.Quic.Implementations.MsQuic.Internal.MsQuicApi::cctor
 * System.Net.Quic.Implementations.MsQuic.Internal.MsQuicApi::IsQuicSupported

... are all involved. Plus the implementation will still try to load a DLL dynamically in MsQuicApi::cctor, which won't exist.

Addressing this inside ASP.NET Core is a more lightweight alternative.

Reply:
@ManickaP In .NET 7 an API should be added to System.Net.Quic that can easily be tested. e.g. [SocketsHttpHandler.IsSupported](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.issupported?view=net-6.0)

Reply:
> Do you know whether it's possible to exclude a property from assembly wide auto generated PNSE?

You can use the `ApiExclusionListPath` property to specify API that's excluded for PNSE generation.  Then you can define that API in a partial class..  Check this commit for an example: https://github.com/dotnet/runtime/commit/cd4157f9024f1cc5c47c4e894842d3e0543f943f